### PR TITLE
feat: typescript support

### PR DESCRIPTION
**Pull request by alvarosabu**
Continuation of #296 

- [x]  module.ts
- [x]  plugin.ts
- [x]  playground.ts

### CODE REVIEW

**Comment by SebbeJohansson** [comment]
🥳🥳🥳🥳

File: playground/pages/index.vue [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<script setup>
+<script setup lang="ts">
 const story = await useAsyncStoryblok("vue", {
   version: "draft",
   language: "en",
-  resolve_relations: ["popular-articles.articles"]
+  resolve_relations: "popular-articles.articles"
```
Comment (Dawntraoz) [inline playground/pages/index.vue:5]: This should be an array, any reason to remove it?

File: lib/src/module.ts [hunk 2]
```diff
@@ -7,21 +7,35 @@ import {
   createResolver
 } from "@nuxt/kit";
 
-export default defineNuxtModule({
+export interface ModuleOptions {
+  accessToken: string,
+  usePlugin: boolean,
+  bridge: boolean, // storyblok bridge on/off
+  apiOptions: any, // storyblok-js-client options
+  useApiClient: boolean,
```
Comment (Dawntraoz) [inline lib/src/module.ts:15]: Can we remove the useApiClient for now, please? It is not being used anymore.

File: lib/src/module.ts [hunk 3]
```diff
@@ -7,21 +7,35 @@ import {
...
     name: "@storyblok/nuxt",
     configKey: "storyblok"
   },
   defaults: {
-    usePlugin: true
+    accessToken: '',
+    usePlugin: true,
+    bridge: true,
+    apiOptions: {},
+    useApiClient: false,
```
Comment (Dawntraoz) [inline lib/src/module.ts:28]: Same here

File: lib/src/module.ts [hunk 4]
```diff
@@ -7,21 +7,35 @@ import {
...
   },
   setup(options, nuxt) {
     const resolver = createResolver(import.meta.url);
 
-    nuxt.options.vite.optimizeDeps.include =
-      nuxt.options.vite.optimizeDeps.include || [];
-    nuxt.options.vite.optimizeDeps.include.push("@storyblok/nuxt");
-    nuxt.options.vite.optimizeDeps.include.push("@storyblok/vue");
+    if(nuxt.options.vite.optimizeDeps) {
```
Comment (Dawntraoz) [inline lib/src/module.ts:33]: We want always to add the optimizeDeps, even if they're not defined in the first place

**Review by Dawntraoz — CHANGES_REQUESTED** [review]
Overall I think it is nice to see the project finally being converted and taking the first steps. Can you please review my comments, shall be simple things 💜

**Comment by SebbeJohansson** [comment]
Pretty sure i cant make changes directly to this PR so I'm gonna leave it for Alvaro! 🕺

File: playground/pages/index.vue (hunk 1 above)
Comment (alvarosabu) [reply playground/pages/index.vue:5]: There is a problem with the type then because it types as `string` should it be `string[]`

**Review by Dawntraoz — APPROVED** [review]

File: playground/pages/vcalong.vue [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<script setup>
```
Comment (alexjoverm) [inline playground/pages/vcalong.vue:1]: I think it's `lang`

**Pull request merged**